vertx-pg-client/src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java

```diff
@@ -1079,6 +1079,11 @@ private static void binaryEncodeBYTEA(Buffer value, ByteBuf buff) {
   }
 
   private static Buffer binaryDecodeBYTEA(int index, int len, ByteBuf buff) {
+    if (buff.isDirect()) {
```

Comment:
is there a case the ByteBuf is not direct ?

Reply:
Afair, my debug logs showed that the `ByteBuf` is not always direct. I will check that.

Reply:
interesting, can you provide stack traces of both cases ? I'm curious to understand the usage pattern

Reply:
I logged the result of `buff.copy` and the resulting `Buffer` could be either a direct `ByteBuf` or not, I did not really understand the pattern when which one was chosen. However the input was always a direct buffer. Sorry for the confusion. I'll remove the condition.